.github/workflows/CIGuardedTrigger.yml

```diff
@@ -0,0 +1,53 @@
+name: PR File Check and Conditional CI Trigger
+
+on:
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - dev
+
+permissions: {}
+
+jobs:
+  check-files:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    
+    - name: Check changed files and conditionally trigger CI
+      env:
+        base_repo: ${{ github.event.pull_request.base.repo.clone_url }}
+        base_branch: ${{ github.event.pull_request.base.ref }}
+        base_branch_pre: "upstream"
+        diff_branch: ${{ github.event.pull_request.head.ref }}
+        REPO_TOKEN: ${{ secrets.CLI_BOT }}
```

Comment:
Just out of curiosity, is the risk of exposing this token controllable?

Reply:
This is a scoped token; it only has comment permissions for azure-cli and azure-cli-extensions.

Reply:
Got it, thanks~